client/assets/calendar.js

```diff
@@ -0,0 +1,121 @@
+	calendar_string += '\nDTSTART:19700308T020000\nRRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:-0700'
+	calendar_string += '\nTZOFFSETTO:-0800\nTZNAME:PST\nDTSTART:19701101T020000\nRRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU\nEND:STANDARD'
+	calendar_string += '\nEND:VTIMEZONE\n'
+
+	plan.courses.forEach(function(plan_course) {
+		//need to use moments to deal with the bug with the start day of the semester not being first day of the course
+		//start_moment = the starting date of the course and time the course starts each day
+		let start_moment = moment(plan_course.dates[0] + '/' + semester_year)
+		//end_moment = the ending date of the course along with the time this course ends each day
+		let end_moment = moment(plan_course.dates[1] + '/' + semester_year)
+
+		start_moment.hour(Number(plan_course.times[0].slice(0,2)))
+		start_moment.minute(Number(plan_course.times[0].slice(3,5)))
+		end_moment.hour(Number(plan_course.times[1].slice(0,2)))
+		end_moment.minute(Number(plan_course.times[1].slice(3,5)))
+
+		//adjust for 24 hr time
+		if (plan_course.times[0].slice(-2) == 'pm' && start_moment.hours() != 12) {
+			start_moment.add(12, 'hours')
+		}
+		if (plan_course.times[1].slice(-2) == 'pm' && end_moment.hours() != 12) {
+			end_moment.add(12, 'hours')
+		}
+
+		//create an array that maps the days proveded by banner to the days needed by ICS standard
+		let course_dows = []
+		plan_course.days.forEach(function(day) {
+			switch(day) {
+				case 'M':
+				course_dows.push(1)
+			break;
+				case 'T':
+				course_dows.push(2)
+			break;
+				case 'W':
+				course_dows.push(3)
+			break;
+				case 'R':
+				course_dows.push(4)
+			break;
+				case 'F':
+				course_dows.push(5)
+			break;
+				case 'S':
+				course_dows.push(6)
+			break;
+				case 'U':
+				course_dows.push(0)
+			break;
+		}
+		})
+
+		//there is an issue with the first day of the semester being later than when the first day of this class is
+		while(!course_dows.includes(start_moment.day())) {
+		start_moment.add(1, 'day')
+		}
+
+		let right_now = (moment().year()).toString() //used to show when this was made in correct ICS format
+		right_now += ((moment().month() + 1).toString().length == 2) ? moment().month() + 1 : '0' + (moment().month() + 1)
+		right_now += (moment().date().toString().length == 2) ? moment().date() : '0' + moment().date()
+		right_now += 'T'
+		right_now += (moment().hours().toString().length == 2) ? moment().hours() : '0' + moment().hours()
+		right_now += (moment().minutes().toString().length == 2) ? moment().minutes() : '0' + moment().minutes()
+		right_now += '00Z'
+
+		//set up timezone info for this event
+		calendar_string += 'BEGIN:VEVENT\nDTSTART;TZID=America/Los_Angeles:'
+		calendar_string += start_moment.year()
+		calendar_string += ((start_moment.month() + 1).toString().length == 2) ? start_moment.month() + 1 : '0' + (start_moment.month() + 1)
```

Comment:
The same applies here: use the library to output in the format you want. no need to do all this.